gordon_gcp/schema/parse.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Module to parse loaded JSON messages according to GCP-related schemas
+and return only the actionable data.
+
+To use:
+
+.. code-block:: pycon
+
+    >>> import json
+    >>> from gordon_gcp.schema import parse
+    >>> parser = parse.MessageParser()
+    >>> # using an example file
+    >>> exp = 'gordon_gcp/schema/examples/audit-log.last-operation.json'
+    >>> with open(exp, 'r') as r:
+    ...   data = json.load(f)
+    >>> message = parser.parse(data, 'audit-log')
+    >>> message
+    {
+        'action': 'additions',
+        'resourceName': 'projects/.../instances/an-instance-name-b45c',
+        'resourceRecords': []
+    }
+
+"""
+
+
+__all__ = ('MessageParser',)
+
+
+class MessageParser:
+    """Parse a message provided a given GCP schema."""
+    ACTION_MAPPER = {
+        'v1.compute.instances.insert': 'additions',
+        'v1.compute.instances.delete': 'deletions',
+    }
+
+    def _parse_audit_log_msg(self, message):
+        # TODO (lynn): what about first vs last messages?
+        payload = message['protoPayload']
```

Comment:
Don't we lose some info here by only taking the proto payload, like timestamp? Or would that be handled outside of this parser?

Reply:
I wasn't sure exactly how timestamp would be used quite yet as I'm writing this consumer, but I can add it in with the anticipation of eventually using it. Doesn't matter to me.